### PR TITLE
 UniFFI & Python - Part 3 of 3 

### PR DESCRIPTION
**Pull request by badboy**
This brings everything back in line:

* tests pass
* upload works
* the general API works

There are still some quirks left, commented code and skipped tests. Those will be tackled afterwards in some cleanups.

### CODE REVIEW

File: glean-core/python/tests/test_glean.py [hunk 1]
```diff
@@ -303,16 +315,19 @@ def test_ping_collection_must_happen_after_currently_scheduled_metrics_recording
...
+            name="string_metric",
+            send_in_pings=[ping_name],
+            dynamic_label=None,
+        )
     )
 
     # This test relies on testing mode to be disabled, since we need to prove the
     # real-world async behaviour of this.
-    Dispatcher._testing_mode = False
+    # Dispatcher._testing_mode = False
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:330]: Was leaving this commented out intentional?

File: glean-core/python/tests/test_glean.py [hunk 2]
```diff
@@ -322,7 +337,7 @@ def test_ping_collection_must_happen_after_currently_scheduled_metrics_recording
     ping.submit()
 
     # Wait until the work is complete
-    Dispatcher._task_worker._queue.join()
+    # Dispatcher._task_worker._queue.join()
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:340]: Another commented out line that I'm not sure if it was intentionally left or not.

File: glean-core/python/tests/test_glean.py [hunk 3]
```diff
@@ -706,8 +690,8 @@ def test_presubmit_makes_a_valid_ping(tmpdir, ping_schema_url, monkeypatch):
 
     # This test relies on testing mode to be disabled, since we need to prove the
     # real-world async behaviour of this.
-    Dispatcher._testing_mode = False
-    Dispatcher._queue_initial_tasks = True
+    # Dispatcher._testing_mode = False
+    # Dispatcher._queue_initial_tasks = True
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:694]: If these need to stay, can we leave a comment as to why?

File: glean-core/python/tests/test_glean.py [hunk 4]
```diff
@@ -723,7 +707,7 @@ def test_presubmit_makes_a_valid_ping(tmpdir, ping_schema_url, monkeypatch):
     )
 
     # Wait until the work is complete
-    Dispatcher._task_worker._queue.join()
+    # Dispatcher._task_worker._queue.join()
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:710]: Another commented out line

File: glean-core/python/tests/test_glean.py [hunk 5]
```diff
@@ -773,8 +758,8 @@ def test_flipping_upload_enabled_respects_order_of_events(tmpdir, monkeypatch):
 
     # This test relies on testing mode to be disabled, since we need to prove the
     # real-world async behaviour of this.
-    Dispatcher._testing_mode = False
-    Dispatcher._queue_initial_tasks = True
+    # Dispatcher._testing_mode = False
+    # Dispatcher._queue_initial_tasks = True
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:762]: More commented out lines.

File: glean-core/python/tests/test_glean.py [hunk 6]
```diff
@@ -791,7 +776,7 @@ def test_flipping_upload_enabled_respects_order_of_events(tmpdir, monkeypatch):
     ping.submit()
 
     # Wait until the work is complete
-    Dispatcher._task_worker._queue.join()
+    # Dispatcher._task_worker._queue.join()
```
Comment (travis79) [inline glean-core/python/tests/test_glean.py:779]: Commented out line

File: glean-core/rlb/tests/schema.rs [hunk 7]
```diff
@@ -10,7 +10,7 @@ use serde_json::Value;
 
 //use glean::private::{DenominatorMetric, NumeratorMetric, RateMetric};
 use glean::net::UploadResult;
-use glean::{ClientInfoMetrics, CommonMetricData, Configuration};
+use glean::{ClientInfoMetrics, /*CommonMetricData,*/ Configuration};
```
Comment (travis79) [inline glean-core/rlb/tests/schema.rs:13]: More comment lint

**Review by travis79 — APPROVED** [review]
Just a handful of nits about some leftover commented out code, nothing to block on! 🎉

File: glean-core/rlb/tests/schema.rs (hunk 7 above)
Comment (badboy) [reply glean-core/rlb/tests/schema.rs:13]: Getting the linter to shut up. Really I should just leave it in until I removed the corresponding use (currently commented out)

File: glean-core/python/tests/test_glean.py (hunk 1 above)
Comment (badboy) [reply glean-core/python/tests/test_glean.py:330]: And here I thought I removed them.

**Pull request merged**